src/foam/nanos/boot/Boot.java

```diff
@@ -23,11 +21,15 @@
   protected X   root_ = new ProxyX();
 
   public Boot() {
-    // Used for all the services that will be required when Booting
-    MapDAO serviceDAO = new MapDAO();
-    serviceDAO.setOf(NSpec.getOwnClassInfo());
-    serviceDAO.setX(root_);
-    serviceDAO_ = serviceDAO;
+    try {
+      // Used for all the services that will be required when Booting
+      DAO nSpecDAO = new MapDAO();
+      ((MapDAO) nSpecDAO).setOf(NSpec.getOwnClassInfo());
+      ((MapDAO) nSpecDAO).setX(root_);
+      serviceDAO_ = new JournaledDAO(nSpecDAO, "services");
```

Comment:
I should be able to just replace all of this with:
DAO nSpecDAO = new JDAO(NSpec, "services");